# Comparative genomics of *Hox* and *ParaHox* genes among major lineages of Branchiopoda with emphasis on tadpole shrimps â€” Data and code
**[Filippo Nicolini](https://linktr.ee/filo.nico/)**, **[Jacopo Martelossi](https://github.com/jacopoM28/)**, **[Giobbe Forni](https://github.com/for-giobbe/)**, **Castrense Savojardo**, **Barbara Mantovani**, **[Andrea Luchetti](https://github.com/andluche)**

In this repository you will find raw data and codes used to perform the analyses of *Hox* and *ParaHox* genes (HPHGs) in Branchiopoda which are found in:

Nicolini F, Martelossi J, Forni G, Savojardo C, Mantovani B and Luchetti A (2023). [Comparative genomics of *Hox* and *ParaHox* genes among major lineages of Branchiopoda with emphasis on tadpole shrimps](https://www.frontiersin.org/articles/10.3389/fevo.2023.1046960/full). *Front. Ecol. Evol.* 11:1046960.

![Immagine 2022-09-02 173353](https://user-images.githubusercontent.com/72141380/188184557-dcbc7ce9-8222-4362-9dca-87f2d16e1952.png)

## Infos
- <code>01_HPHG_seqs/</code> contains HPHG sequences in fasta format and also data and R script to generate the raw version of **Suppl. Fig. S2**.
- <code>02_HPHG_trees/</code> contains HPHG ML phylogenetic trees as in **Fig. 1A** and **Fig. 3**.
- <code>03_HPHG_structure/</code> contains data and R script to generate the raw version of **Fig. 1B**.
- <code>04_HPHG_statistics/</code> contains data and R script to generate the raw version of **Fig. 2**.
- <code>05_RRTree/</code> contains data and command files to perform the RRTree analysis.
- <code>06_HyPhy/</code> contains data and command files to perform the HyPhy analysis.
- <code>07_HPHG_gffs/</code> contains annotation files of HPHG in branchiopods.
- <code>scripts/</code> contains additional bash scripts that I wrote to manage HPHG analyses during my internship.
